util/postprocess-makedepend.pl

```diff
@@ -0,0 +1,75 @@
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+my $producer = $ARGV[0];
+
+while (<STDIN>) {
+    if ($producer eq 'makedepend') {
+
+        # makedepend, in its infinite wisdom, wants to have the object file in
+        # the same directory as the source file.  This doesn't work too well
+        # with out-of-source-tree builds, so we must resort to tricks to get
+        # things right.  The trick is to call makedepend with an extra suffix
+        # that contains the desired object file path, like this:
+        #
+        #     makedepend -f- -o"|dir/foo.o" -- $(CFLAGS) -- ../somewhere/foo.c
+        #
+        # The result will look something like this:
+        #
+        #     ../somewhere/foo|dir/foo.o: deps...
+        #
+        # Which is easy to massage by removing everything up to the first |
+
+        # Remove everything up to the first |
+        s/^.*\|//;
+        # Also, remove any dependency that starts with a /, because those are
+        # typically system headers
+        s/ \/(\\.|[^ ])*//;
+        # Finally, remove all empty lines or comment lines
+        $_ = undef if (/: *$/ || /^(#.*| *)$/);
+        $_.="\n" unless !defined($_) or /\R$/g;
+
+    } elsif ($producer eq 'VMS C') {
+
+        # current versions of DEC / Compaq / HP / VSI C strips away all
+        # directory information from the object file, so we must insert it
+        # back. Just to be safe against future changes, we check that there
+        # really is no directory information.
+
+        my $directory = $ARGV[1];
+
+        # The pattern for target and dependencies will always take this form:
+        #
+        #     target SPACE : SPACE deps
+        #
+        # This is so a volume delimiter (a : without any spaces around it)
+        # won't get mixed up with the target / deps delimiter.  We use this
+        # fact in the regexp below to make sure we do look at the target.
+        s/^/$directory/ unless /^\S+[:<\]]\S+\s+:/;
+
+    } elsif ($producer eq 'Windows C') {
+
+        # For the moment, we only know of one native Windows C compiler, and
+        # that's Visual C.  With that compiler, the flags /Zs /showIncludes
+        # give us the necessary output to be able to create dependencies that
+        # nmake (or any 'make' implementation) should be able to read, with a
+        # bit of help.  The output we're interested in looks like this (it
+        # always starts the same)
+        #
+        #     Note: including file: {whatever header file}
+        #
+        # So all we really have to do is to is to replace the start of the line
+        # with an object file specification, given to us as $ARGV[1].
+
+        my $object = $ARGV[1];
+        s/^Note: including file: */${object}: /;
+
+    }
```

Comment:
I think there should be an explicit no-op which just copies things or an error if it's not one of the supported types.

Reply:
An error would be the better choice, methinks, 'cause that's would be on us.  Good point, will fix.